added german translation to i18n

NoahGysin opened the pull request:
<!-- Please don't delete this template -->

<!-- PULL REQUEST TEMPLATE -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

## Summary

Added german translation to the i18n module in the file "de.json".

## What are the specific steps to test this change?

> 1. Run the website and log in as an admin
> 2. Change the language of the CMS to german

## What kind of change does this PR introduce?

- [ ] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [x] Other

BoDonkey commented on modules/@apostrophecms/i18n/i18n/de.json:
```diff
@@ -0,0 +1,427 @@
+  "archivePageAndSubpages": "Seite mit allen Unterseiten archivieren",
+  "archiveType": "{{ type }} archivieren",
+  "archiveTypeAffirmativeLabel": "Ja, {{ type }} archivieren",
+  "archiveTypeNote": "Du schaust dir im Moment den {{ type }} an den du archivieren willst. Wenn die Archivierung abgeschlossen ist wirst du zur Startseite zurückgeleitet.",
```
I just happened to catch this. There may be some other grammatical changes needed. Can you give one more look-through, please?

NoahGysin replied:
Ah good catch, yeah I've got some time on my hands today so I will go over all the translations again and iron out the last grammatical errors.

NoahGysin commented on modules/@apostrophecms/i18n/i18n/de.json:
```diff
@@ -0,0 +1,427 @@
+  "cancel": "Abbrechen",
+  "cannotMoveArchive": "Das Archiv kann nicht verschoben werden.",
+  "changed": "Geändert",
+  "changesAwaitingApproval": "Änderungen warten auf Freigabe durch einen Administrator oder Redakteur.",
```
That's true, I've made the translation over a couple of days so that probably led to those inconsistencies. Will change that as well ;)

NoahGysin commented:
Alrighty went through the whole file again and fixed the grammaticals errors. I also found some new lines added in the english file which I added and translatet to the german translation as well.

BoDonkey reviewed (dismissed):
Looks great! Thank you very much for doing this and taking the time to go back through your translations. I'm merging it to main and it will be released this week. 🎉

BoDonkey reviewed (approved):
Resolves README conflicts.
